invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentMetadata/DocumentMetadata.js

```diff
@@ -30,39 +41,89 @@ export default class DocumentMetadata extends Component {
     );
   }
 
-  render() {
-    const document = this.props.documentDetails;
-    const rows = [
-      { name: 'Title', value: document.metadata.title },
-      { name: 'Authors', value: document.metadata.authors },
-    ];
-    if (!_isEmpty(document.metadata.keywords)) {
-      rows.push({
-        name: 'Keywords',
-        value: this._renderKeywords(document.metadata.keywords),
-      });
-    }
-    const header = (
+  handleOnRefClick(loanPid) {
+    const navUrl = BackOfficeRoutes.loanDetailsFor(loanPid);
+    window.open(navUrl, `_loan_${loanPid}`);
+  }
+
+  createRefProps(documentPid) {
+    const loanStates = invenioConfig.circulation.loanActiveStates;
+    loanStates.push('PENDING');
+
+    const loanRefProps = {
+      refType: 'Loan',
+      onRefClick: this.handleOnRefClick,
+      getRefData: () =>
```

Comment:
An example of the function that returns the promise, referring to https://github.com/inveniosoftware/invenio-app-ils/pull/335/files#r282867373